ncclient/operations/third_party/iosxe/rpc.py

```diff
@@ -1,9 +1,500 @@
+            self.parse()
+        return self._subscription_result.text
+
+    @property
+    def subscription_result_ele(self):
+        "*subscription-result* element as an :class:`~xml.etree.ElementTree.Element`"
+        if not self._parsed:
+            self.parse()
+        return self._subscription_result
+
+    @property
+    def subscription_result_xml(self):
+        "*subscription-result* element as an XML string"
+        if not self._parsed:
+            self.parse()
+        return to_xml(self._subscription_result)
+
+
+class DeleteSubscription(RPC):
+    
+    "`establish-subscription` RPC"
+
+    DEPENDS = ['urn:ietf:params:netconf:capability:notification:1.1']
+    REPLY_CLS = DeleteSubscriptionReply
+    
+    def request(self, subscription_id=None):
+        """Create a simple subscription for ietf-yang-push subscriptions.
+
+        *subscription_id* the id of the subscription to delete
+
+        """
+        #
+        # validate parameters
+        #
+        if subscription_id is None:
+            raise YangPushError("Must provide subscription_id")
+
+        #
+        # Construct request
+        #
+        node = new_ele_ns("delete-subscription", IETF_EVENT_NOTIFICATIONS_NS)
+        to_delete = sub_ele_ns(node, "subscription-id", IETF_EVENT_NOTIFICATIONS_NS)
+        to_delete.text = str(subscription_id)
+
+        # remove subscription-id callbacks
+        self.session.yang_push_listener.remove_subscription_listener(int(subscription_id))
+        
+        # Now process the request
+        return self._request(node)
+
+
+class YangPushNotificationType(object):
+
+    """Simple enumeration of YANG push notification types."""
+    
+    UNKNOWN = 0
+    PUSH_UPDATE = 1
+    PUSH_CHANGE_UPDATE = 2
+
+    @staticmethod
+    def str_to_type(string):
+        lookup = {
+            "push-update": YangPushNotificationType.PUSH_UPDATE,
+            "push-change-update": YangPushNotificationType.PUSH_CHANGE_UPDATE,
+        }
+        try:
+            return lookup[string]
+        except:
+            raise Exception("Unknown YANG push notification type")
+
+
+class YangPushNotification(object):
+
+    """Represents a YANG Push `notification`."""
+    
+    def __init__(self, raw):
+        self._raw = raw
+        self._parsed = False
+        self._root = None
+        self._datastore = None
+        self._event_time = None
+        self._subscription_id = None
+        self._type = None
+        self._invalid = False
+        
+    def __repr__(self):
+        return self._raw
+    
+    def parse(self):
+        try:
+            root = self._root = to_ele(self._raw)
+
+            # extract eventTime
+            event_time = root.find(qualify("eventTime", NETCONF_NOTIFICATION_NS))
+            if event_time is not None:
+                self._event_time = parse(event_time.text)
+
+            # determine type of event
+            type = root.find(qualify("push-update", IETF_YANG_PUSH_NS))
```

Comment:
Nitpick: "type" is a python keyword, so I'm not a big fan of using it as a variable name. I'd suggest "update_type" or "notif_type" instead.